### PR TITLE
Bootstrap 3 Migration: Messaging > Chat

### PR DESCRIPTION
**Pull request by pnaw94**
@biyeun 
scr01

### CODE REVIEW

File: corehq/apps/reports/static/reports/javascripts/config.dataTables.bootstrap.js [hunk 1]
```diff
@@ -84,12 +85,16 @@ function HQReportDataTables(options) {
...
                 bPaginate: self.show_pagination,
-                sPaginationType: self.paginationType,
                 iDisplayLength: self.defaultRows,
                 bAutoWidth: self.autoWidth,
                 sScrollX: "100%",
                 bSort: self.defaultSort
             };
+
+            if(!self.useBootstrap3) {
+                params['sPaginationType'] = self.paginationType
```
Comment (dimagimon) [inline corehq/apps/reports/static/reports/javascripts/config.dataTables.bootstrap.js]: ['sPaginationType'] is better written in dot notation.

Comment (dimagimon) [inline corehq/apps/reports/static/reports/javascripts/config.dataTables.bootstrap.js]: Missing semicolon.

**Comment by pnaw94** [comment]
By the way, I have a feeling that 'datatables-bootstrap3/BS3/assets/css/datatables.css' autoincluded (or at least some part of it) does collide with HQReportDataTable as having the following styling:
```
div.dataTables_length select {
	width: 75px;
}
```
...while on the other hand in 'config.dataTables.bootstrap.js' we insert some additional text to page lenght select options:
`$selectField.children().append(" per page");`

...produces an ugly-looking select that does not match it's text length and for every table it's necessary to insert some additional css in order to override original styling.

File: corehq/apps/reports/static/reports/javascripts/config.dataTables.bootstrap.js [hunk 2]
```diff
@@ -9,6 +9,7 @@ function HQReportDataTables(options) {
     self.startAtRowNum = options.startAtRowNum || 0;
     self.showAllRowsOption = options.showAllRowsOption || false;
     self.aoColumns = options.aoColumns;
+    self.aoColumnDefs = options.aoColumnDefs;
```
Comment (biyeun) [inline corehq/apps/reports/static/reports/javascripts/config.dataTables.bootstrap.js]: passing 'bootstrap' as the option for this is actually how this should be handled.

**Comment by biyeun** [comment]
ok, there are a lot more modifications to `config.dataTables.bootstrap.js` than i expected for this PR. This has a much larger effect across all reports, so I'm going to handle this separately and cherry pick changes. Closing.

**Pull request closed without merging**